src/Adapter/SymfonyCache.php

```diff
@@ -169,6 +178,17 @@ public function cacheAction(string $token, string $type): Response
 
         $path = 'all' === $type ? $this->cacheDir : sprintf('%s/%s', $this->cacheDir, $type);
 
+        // clean up event dispatcher just before clearing the cache to prevent further events execution
+        if (null !== $this->eventDispatcher) {
```

Comment:
If it is null, please trigger a deprecation error stating that the argument will be mandatory in the upcoming release.